Source/WebCore/Modules/mediastream/MediaStreamTrack.cpp

```diff
@@ -328,6 +330,22 @@ void MediaStreamTrack::getPhotoCapabilities(DOMPromiseDeferred<IDLDictionary<Pho
     });
 }
 
+void MediaStreamTrack::getPhotoSettings(DOMPromiseDeferred<IDLDictionary<PhotoSettings>>&& promise) const
+{
+    m_private->getPhotoSettings()->whenSettled(RunLoop::main(), [protectedThis = Ref { *this }, promise = WTFMove(promise)] (RealtimeMediaSource::PhotoSettingsPromise::Result&& result) mutable {
+        if (!result) {
+            // https://w3c.github.io/mediacapture-image/#ref-for-dom-imagecapture-getphotosettings②
+            // If the data cannot be gathered for any reason (for example, the MediaStreamTrack being ended
+            // asynchronously), then reject p with a new DOMException whose name is OperationError, and
```

Comment:
It would be good to add a test case like this one: call getPhotoSettings(), end the track and check the results.
It is not totally clear what we should do here, probably reject would be best but the current patch does not guarantee it in its current form.

Reply:
It would also be good to add a test where we create a RTCPeerConnection based track, in which case we will probably get an OperationError with `not supported error` message, which is up to spec I think but a bit strange (it might be somehow better to have NotSupportedError).